docs/root/intro/arch_overview/http/http_routing.rst

```diff
@@ -14,12 +14,8 @@ level the router takes an incoming HTTP request, matches it to an upstream clust
 request. The router filter supports the following features:
 
 * Virtual hosts that map domains/authorities to a set of routing rules.
-* Prefix and exact path matching rules (both :ref:`case sensitive
-  <envoy_v3_api_field_config.route.v3.RouteMatch.case_sensitive>` and case insensitive). Regex/slug
-  matching is not currently supported, mainly because it makes it difficult/impossible to
-  programmatically determine whether routing rules conflict with each other. For this reason we
-  don’t recommend regex/slug routing at the reverse proxy level, however we may add support in the
-  future depending on demand.
+* Prefix and exact path matching rules (both :ref:`case sensitive <envoy_v3_api_field_config.route.v3.RouteMatch.case_sensitive>` and case insensitive). 
+* :ref:`Regex path matching <envoy_v3_api_field_config.route.v3.RouteMatch.safe_regex` rules.
```

Comment:
hi @saiskee the PR is failing due to trailing whitespace, could you clean up and push again

Reply:
its also failing due missing closing bracket

```
undefined label: regex path matching <envoy_v3_api_field_config.route.v3.routematch.safe_regex
```

Reply:
Thanks @phlax, are there any tools for formatting docs similar to clang?

Reply:
im working on it right now

the first issue would be picked up by running `./tools/code_format/glint.sh`

unfortunately with the second issue i think you would actually need to build the docs at least atm - im working on adding `rstcheck` which i would hope to pick it up - not 100% sure tho

Reply:
re the first issue - i would recommend setting your editor to automatically fixing the problems that `./glint.sh` picks up, ie

- no trailing whitespace
- newline (not blank line) at end of all files
- no mixed tab and space indents

Reply:
re IDE auto fixing problems - Do I need an extension for this? I am using VsCode.

Reply:
not sure i use emacs

Reply:
https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut